src/Message.php

```diff
@@ -86,7 +91,12 @@ public function addHeader(string $name, $value) {
         \assert($this->isNameValid($name), "Invalid header name");
 
         if (\is_array($value)) {
-            $value = \array_map("strval", $value);
+            if (!$value) {
+                $this->removeHeader($name);
```

Comment:
The header should not be removed from this method, this should just `return`.

Reply:
Oops, yes. Missed that.

Reply:
Fixed via acb8019923f4e8fed6abcff1fdf2dae9849f5c54.